docs/source/daml/intro/daml/3_Data/Keys.daml

```diff
@@ -0,0 +1,86 @@
+daml 1.2
+module Keys where
+
+-- KEY_TEST_BEGIN
+data Bank = Bank with
+  party : Party
+  address: Text
+  telephone : Text
+    deriving (Eq, Show)
+
+data AccountKey = AccountKey with
+  accountant : Party
+  number : Text
+  bank_party : Party
+    deriving (Eq, Show)
+
+template Account
+  with
+    accountant : Party
+    owner : Party
+    number : Text
+    bank : Bank
+  where
+    signatory accountant
+
+    key AccountKey with
+        accountant
+        number
+        bank_party = bank.party
+      : AccountKey
+    maintainer accountant
+
+data Cash = Cash with
+  currency : Text
+  amount : Decimal
+    deriving (Eq, Show)
+
+template CashBalance
+  with
+    accountant : Party
+    cash : Cash
+    account : AccountKey
+  where
+    signatory accountant
+
+id_ref_test = scenario do
+  accountant <- getParty "Bob"
+  owner <- getParty "Alice"
+  bank_party <- getParty "Bank"
+  let
+    bank = Bank with
+      party = bank_party
+      address = "High Street"
+      telephone = "012 3456 789"
+    cash = Cash with
+      currency = "USD"
+      amount = 100.0
+
+  accountCid <- submit accountant do
+     create Account with
+      accountant
+      owner
+      bank
+      number = "ABC123"
+
+  balanceCid <- submit accountant do
+    account <- fetch accountCid
+    create CashBalance with
+      accountant
+      cash
+      account = key account
+
+  -- Now the accountant updates the telephone number for the bank on the account
+  new_accountCid <- submit accountant do
+    account <- fetch accountCid
+    archive accountCid
+    create account with
+      bank = account.bank with
+        telephone = "098 7654 321"
+
+  -- The `account` field on the balance is now stale.
```

Comment:
Isn't the whole point of using the key instead of the contractID that we get the correct value now, rather than a stale one?

Reply:
indeed. fixing.